packages/product/src/services/product-module-service.ts

```diff
@@ -271,107 +248,127 @@ export default class ProductModuleService<
 
   @InjectTransactionManager("baseRepository_")
   protected async updateVariants_(
-    data: ProductTypes.UpdateProductVariantOnlyDTO[],
+    data: ProductTypes.UpdateProductVariantDTO[],
     @MedusaContext() sharedContext: Context = {}
   ): Promise<TProductVariant[]> {
+    // Validation step
```

Comment:
suggestion: Look into improving performance of this method by replacing arrays with maps. We are looping over the same data a couple of times, I believe – improvements could cascade to `diffVariants`.

Note, I usually don't care much about this because often times we would be optimizing for an ultra-edge case, but variants is actually one of those places where it makes sense to be meticulous about achieving good performance, as we've seen many cases with dozens and even hundreds of variants on products.

This can be punted for a follow-up itereation, so this doesn't hang any longer.

Reply:
I think there are a lot of opportunities for performance improvements, but most of them would be at the DB level as that's where most of the time is spent. Using maps is also nice of course, but usually not where bottlenecks are. 

Anyway this module needs a lot more work, for now I'll focus on functionality first.